File: backend/shared/uidl/src/main/java/io/mateu/uidl/data/Badge.java
package io.mateu.uidl.data;

import io.mateu.uidl.fluent.Component;
import lombok.Builder;

@Builder
public record Badge(
    String text,
    String iconOnLeft,
    String iconOnRight,
    BadgeColor color,
    boolean primary,
    boolean small,
    boolean pill,
    String style,
    String cssClasses)
    implements Component {

  public Badge(String text) {
    this(text, "", "", null, false, false, false, "", "");
  }

  @Override
  public BadgeColor color() {
    return color != null ? color : BadgeColor.normal;
  }
}

File: frontend/web/vaadin/src/mateu/shared/apiClients/dtos/componentmetadata/Badge.ts
import { BadgeColor } from "./BadgeColor";
import ComponentMetadata from "@mateu/shared/apiClients/dtos/ComponentMetadata";

export default interface Badge extends ComponentMetadata {

    text: string
    iconOnLeft: string
    iconOnRight: string
    color: BadgeColor
    primary: boolean
    small: boolean
    pill: boolean

}
